### ISSUE TITLE
Everything works but no recordings show up

### ISSUE THREAD

**Issue by sleblanc** [MythTV-Clients/MythTV-Android-Frontend#107]
All other functions seem to work - so I see the guide and upcoming recordings - just no existing recordings. The troubleshooting steps also show no messages at all for any operation - e in though other operations work.

Any ideas?

**Comment by billmeek** [comment]
If you're referring to "no messages at all" in the mythbackend logs, did
you do the: mythbackend --setverbose upnp command first?

**Comment by sleblanc** [comment]
Thanks - that gave me  the logs.

When I hit Recordings on the tablet - I get the following (which I think 
is corretc) - but no recording info is returned!
Any ideas?

<pre>
Jan 17 23:14:37 mythserver mythbackend[22939]: I HttpServer42 
httprequest.cpp:1281 (ExtractMethodFromURL) ExtractMethodFromURL(end) :  : /
Jan 17 23:14:37 mythserver mythbackend[22939]: I HttpServer42 
httprequest.cpp:228 (SendResponse) HTTPRequest::SendResponse( File ) 
:200 OK -> 192.168.0.109:
Jan 17 23:14:37 mythserver mythbackend[22939]: I HttpServer42 
httprequest.cpp:292 (SendResponseFile) SendResponseFile ( 
/usr/share/mythtv/html/index.html )
Jan 17 23:14:37 mythserver mythbackend[22939]: I HttpServer42 
httprequest.cpp:834 (TestMimeType) 
HTTPRequest::TestMimeType(/usr/share/mythtv/html/index.html) - type is 
text/html
Jan 17 23:14:37 mythserver mythbackend[22939]: I HttpServer42 
httprequest.cpp:518 (SendFile) SendResponseFile : --- size = 0, offset = 
3021, sent = 3021
Jan 17 23:14:38 mythserver mythbackend[22939]: I HttpServer42 
httprequest.cpp:1281 (ExtractMethodFromURL) ExtractMethodFromURL(end) : 
GetRecordedList : /Dvr
Jan 17 23:14:38 mythserver mythbackend[22939]: I HttpServer42 
servicehost.cpp:281 (ProcessRequest) ServiceHost::ProcessRequest: 
GetRecordedList : GET /Dvr/GetRecordedList HTTP/1.1#015
Jan 17 23:14:38 mythserver mythbackend[22939]: I HttpServer42 
httprequest.cpp:241 (SendResponse) HTTPRequest::SendResponse(xml/html) 
() :200 OK -> 192.168.0.109: 4
</pre>

**Comment by billmeek** [comment]
Yes, that looks good, most importantly: 

<pre>
Jan 17 23:14:38 mythserver mythbackend[22939]: I HttpServer42 servicehost.cpp:281 (ProcessRequest) ServiceHost::ProcessRequest: GetRecordedList : GET /Dvr/GetRecordedList HTTP/1.1#015
Jan 17 23:14:38 mythserver mythbackend[22939]: I HttpServer42 httprequest.cpp:241 (SendResponse) HTTPRequest::SendResponse(xml/html) () :200 OK -> 192.168.0.109: 4
</pre>

From a browser on your phone/tablet, please enter:

```
http://<yourBackendIP>:6544/Dvr/GetRecordedList ,
```

and you should see XML for all of your recordings. It may be huge (and it's sent on
a single line) so please use a pastebin and put a link to it here.

Also, what version of MythTV Android Frontend are you running? 1.0.4 is the current
release. Please mention your Linux distribution and MythTV version too.

**Comment by sleblanc** [comment]
Thanks again!

Linux distro is Ubuntu

MythTV Version : v0.25.3-31-gfa3ebd4
MythTV Branch : fixes/0.25
Network Protocol : 72
Library API : 0.25.20130112-1
QT Version : 4.7.4

MythTV Android - latest version (1.0.4)

If the URL above really should have the comma - then I get an error - with no comma - I get the XML output.

Here's the link  to the output (which looks good as far as I can tell)

https://gist.github.com/4562542

**Comment by billmeek** [comment]
It does look good (and the comma was a typo.)

Can you describe in some more detail what happens when you go to the
Recordings screen? There should be messages saying:

```
Syncing Recorded Programs and then
Recorded Programs are up to date!
```

And I forgot to ask, what type of Android device are you using?

**Comment by sleblanc** [comment]
That's exactly what I get - just no recordlings in the list.

|Syncing Recorded Programs and then
Recorded Programs are up to date!|

I running on a Nexus 7 with Android 4.2.1

**Comment by dmfrey** [comment]
Can you copy and send the output of the GetRecordedList call previously
mentioned in this thread?  I am thinking it is failing to parse the XML for
some reason.  Possibly with a special character or something in the XML
getting returned.  You should be able to attach it here as a text document.

On Fri, Jan 18, 2013 at 8:45 AM, sleblanc notifications@github.com wrote:

> That's exactly what I get - just no recordlings in the list.
> 
> |Syncing Recorded Programs and then
> Recorded Programs are up to date!|
> 
> I running on a Nexus 7 with Android 4.2.1
> 
> On 1/18/2013 1:28 AM, billmeek wrote:
> 
> > It does look good (and the comma was a typo.)
> > 
> > Can you describe in some more detail what happens when you go to the
> > Recordings screen? There should be messages saying:
> > 
> > |Syncing Recorded Programs and then
> > Recorded Programs are up to date!
> > |
> > 
> > And I forgot to ask, what type of Android device are you using?
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > <
> > https://github.com/MythTV-Clients/MythTV-Android-Frontend/issues/107#issuecomment-12409395>.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/MythTV-Clients/MythTV-Android-Frontend/issues/107#issuecomment-12422248.

## 

My Google Profile http://www.google.com/profiles/dmfrey

**Comment by sleblanc** [comment]
There was already a link in the ticket to the output. Here is is again ...

https://gist.github.com/4562542

Steve

**Comment by dmfrey** [comment]
That data is incomplete.  There are only 2 lines in the document posted.
 Can you re-run it again and re-attach it?

How many recordings do you have on your backend presently?

On Fri, Jan 18, 2013 at 9:00 AM, sleblanc notifications@github.com wrote:

> There was already a link in the ticket to the output. Here is is again ...
> 
> https://gist.github.com/4562542
> 
> Steve
> 
> On 1/18/2013 8:50 AM, Daniel Frey wrote:
> 
> > Can you copy and send the output of the GetRecordedList call previously
> > mentioned in this thread? I am thinking it is failing to parse the XML
> > for
> > some reason. Possibly with a special character or something in the XML
> > getting returned. You should be able to attach it here as a text
> > document.
> > 
> > On Fri, Jan 18, 2013 at 8:45 AM, sleblanc notifications@github.com
> > wrote:
> > 
> > > That's exactly what I get - just no recordlings in the list.
> > > 
> > > |Syncing Recorded Programs and then
> > > Recorded Programs are up to date!|
> > > 
> > > I running on a Nexus 7 with Android 4.2.1
> > > 
> > > On 1/18/2013 1:28 AM, billmeek wrote:
> > > 
> > > > It does look good (and the comma was a typo.)
> > > > 
> > > > Can you describe in some more detail what happens when you go to the
> > > > Recordings screen? There should be messages saying:
> > > > 
> > > > |Syncing Recorded Programs and then
> > > > Recorded Programs are up to date!
> > > > |
> > > > 
> > > > And I forgot to ask, what type of Android device are you using?
> > > > 
> > > > —
> > > > Reply to this email directly or view it on GitHub
> > > > <
> > 
> > https://github.com/MythTV-Clients/MythTV-Android-Frontend/issues/107#issuecomment-12409395
> > .
> > 
> > > —
> > > Reply to this email directly or view it on
> > > GitHub<
> > > https://github.com/MythTV-Clients/MythTV-Android-Frontend/issues/107#issuecomment-12422248
> > > .
> > 
> > ## 
> > 
> > My Google Profile http://www.google.com/profiles/dmfrey
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > <
> > https://github.com/MythTV-Clients/MythTV-Android-Frontend/issues/107#issuecomment-12422446>.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/MythTV-Clients/MythTV-Android-Frontend/issues/107#issuecomment-12422809.

## 

My Google Profile http://www.google.com/profiles/dmfrey

**Comment by billmeek** [comment]
@dmfrey, the 2nd line has all the data, it's just all on the 2nd line. That has
been my experience any time I've done this before. It's 384kB with 219
programs.

To look at it, I use vi and then do:

```
 :g/>/s//>^V^M/g
```

It won't have the  proper indenting, but at least it can be searched. I'll look
at his again with an eye out for special characters. Good tip.

**Comment by billmeek** [comment]
I noticed that @sleblanc has the following:

```
...
<Channel>
    <ChanId>1021</ChanId>
    <ChanNum>#1021</ChanNum>
    <CallSign>#1021</CallSign>
    <IconURL/>
    <ChannelName>#1021</ChannelName>
...
```

My ChanNum/CallSign/ChannelName have no # sign in them (and use things
like 5_1/WXYZ/WZYZ respectively.)

I did a g/[a-z]/s///g to remove all lower case ASCII, then repeated with [A-Z]
and [0-9]. Then deleted /^<>$/ and ^<\/>$ etc. That left the file with mostly
punctuation.

 I didn't see any other characters that struck me as out of place.

**Comment by dmfrey** [comment]
We also need to see the result of GetChannelList to see if these match.

On Fri, Jan 18, 2013 at 11:22 AM, billmeek notifications@github.com wrote:

> I noticed that @sleblanc https://github.com/sleblanc has the following:
> 
> ...
> <Channel>
>     <ChanId>1021</ChanId>
>     <ChanNum>#1021</ChanNum>
>     <CallSign>#1021</CallSign>
>     <IconURL/>
>     <ChannelName>#1021</ChannelName>
> ...
> 
> My ChanNum/CallSign/ChannelName have no # sign in them (and use things
> like 5_1/WXYZ/WZYZ respectively.)
> 
> I did a g/[a-z]/s///g to remove all lower case ASCII, then repeated with
> [A-Z]
> and [0-9]. Then deleted /^<>$/ and ^<\/>$ etc. That left the file with
> mostly
> punctuation.
> 
> I didn't see any other characters that struck me as out of place.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/MythTV-Clients/MythTV-Android-Frontend/issues/107#issuecomment-12429248.

## 

My Google Profile http://www.google.com/profiles/dmfrey

**Comment by billmeek** [comment]
@sleblanc, please pastebin the output of:

```
http://<yourBackendIP>:6544/Channel/GetChannelInfoList
```

Side note, when I do the above (0.27pre2,) ALL strings are 
empty and integers are 0. Yet the <Count>72</Count> matches
the SQL query SELECT chanid FROM channel;. GetChannelList
doesn't exist.

```
...
<ChannelInfos>
  <ChannelInfo>
    <ChanId>0</ChanId>
    <ChanNum/>
    <CallSign/>
    <IconURL/>
    <ChannelName/>
...
```

**Comment by sleblanc** [comment]
Thanks!
I updated the same document with formatted XML output from the GetChannelInfoList call as asked. I put the results in the same doc as before - and also updated the GetRecorded output to formatted xml (same doc)

As you suggested - the channelList data is all zero!

https://gist.github.com/4562542

**Comment by billmeek** [comment]
I retried the last test and IT WORKED this morning. I can't explain why
other that the host has been powered down between tests, not to mention
dozens of backend restarts. Here's a direct copy/paste of what I used:

```
http://192.168.1.223:6544/Channel/GetChannelInfoList
```

You can add the Source ID, like this too:

```
http://192.168.1.223:6544/Channel/GetChannelInfoList?SourceID=1
```

Please consider restarting your backend in order to get an answer to
@dmfrey's question. Thanks.

**Comment by sleblanc** [comment]
Thanks! I've restarted my back end multiple times - still the same result. One question though - When I perform the second operation - I get results (including a SourceID of 1) - with no source ID specified - I get XML elements for all of the channels - but no data. Does the Android Frontend code to show recordings use this call? When I do a call to Channel/GetVideoSourceList - I get the following (I only have one):

```
<VideoSourceList version="1.0" serializerVersion="1.1">
  <AsOf>2013-01-21T20:16:48</AsOf>
  <Version>0.25.20130112-1</Version>
  <ProtoVer>72</ProtoVer>
  <VideoSources>
    <VideoSource>
      <Id>1</Id>
      <SourceName>SchedulesDirect-Comcast Naples</SourceName>
      <Grabber>schedulesdirect1</Grabber>
      <UserId>REMOVED</UserId>
      <FreqTable>default</FreqTable>
      <LineupId>FL61795:X</LineupId>
      <Password>REMOVED</Password>
      <UseEIT>false</UseEIT>
      <ConfigPath/>
      <NITId>-1</NITId>
    </VideoSource>
  </VideoSources>
</VideoSourceList>
```

Restarting isn't going to change the source_id.

**Comment by billmeek** [comment]
When I looked at the backend code, if SourceID isn't provided,
then a an SQL query like this is made:

```
SELECT chanid FROM channel WHERE sourcied = 0;
```

which returns no data. So, from the backend's perspective,
the SourceID=n is required. I just haven't found any documentation
that says it. Sorry for the bad info. If I understand you last post
correctly, you did get info when SourceID=1 was used. If so,
please post the output where you did last time. Thanks.

**Comment by sleblanc** [comment]
Ok - I updated the same document with the XML output with SourceID set to '1'. If the expectation is sending no source ID in - and you'll get ALL sources back - that appears to not work that way (in fact I'm suprised you get anything back - because I tried sending in a SourceID of 2, and got nothing back.

It would appear that to get all of the channels from all sources - you should execute the 'Channel/GetVideoSourceList ' as above - then iterate over the results with calls to 'Channel/GetChannelInfoList?SourceID={value from Source List}'.

Here's the link to the output:

https://gist.github.com/4562542

**Comment by billmeek** [comment]
Unless you truncated the list:

That says that you only have 26 channels defined. I didn't look any
further, but chanid 1021 isn't in the list, but it is in the programs:

```
        <ChanId>1203</ChanId>
        <ChanId>1204</ChanId>
        <ChanId>1205</ChanId>
        <ChanId>1208</ChanId>
        <ChanId>1212</ChanId>
        <ChanId>1214</ChanId>
        <ChanId>1215</ChanId>
        <ChanId>1804</ChanId>
        <ChanId>1911</ChanId>
        <ChanId>1912</ChanId>
        <ChanId>1498</ChanId>
        <ChanId>1499</ChanId>
        <ChanId>1501</ChanId>
        <ChanId>1508</ChanId>
        <ChanId>1509</ChanId>
        <ChanId>1510</ChanId>
        <ChanId>1536</ChanId>
        <ChanId>1206</ChanId>
        <ChanId>2211</ChanId>
        <ChanId>1213</ChanId>
        <ChanId>1181</ChanId>
        <ChanId>1182</ChanId>
        <ChanId>1211</ChanId>
        <ChanId>1184</ChanId>
        <ChanId>1207</ChanId>
        <ChanId>1183</ChanId>
```

If it were me, I'd verify that on the backend like this:

```
mysql -umythtv -p mythconverg -e \
    "SELECT chanid FROM channel ORDER BY chanid"
```

**Comment by sleblanc** [comment]
That actually makes sense  - cable provider redefined channels so some
channels would be there on old recordings - but no channel data.

I also the imported a bunch of old recordings from a crashed system - it
could also be these that are causing issues.

I did not truncate any data - but can look at adjusting it - would that
cause recordings not to show up?

**Comment by billmeek** [comment]
I'd suggest 'fixing' one channel to see if it makes a difference.
If that works, then they all need to be adjusted. If you have
mythweb working, its easy to do. All the relevant channel
information is on one page. Maybe start with 1021.

**Comment by sleblanc** [comment]
Fixed all the 'old' channels to the 'new' channel numbers (lucky I'm only recording from 3-4 stations at this point) - still no luck! Now - a single recording listing looks like this ...

```
<Program>
  <StartTime>2013-01-22T03:01:00</StartTime>
  <EndTime>2013-01-22T04:00:00</EndTime>
  <Title>Castle</Title>
  <SubTitle>Death Gone Crazy</SubTitle>
  <Category>Crime drama</Category>
  <CatType/>
  <Repeat>false</Repeat>
  <VideoProps>9</VideoProps>
  <AudioProps>9</AudioProps>
  <SubProps>1</SubProps>
  <SeriesId>EP01085588</SeriesId>
  <ProgramId>EP010855880095</ProgramId>
  <Stars>0</Stars>
  <FileSize>6248030352</FileSize>
  <LastModified>2013-01-22T04:37:47</LastModified>
  <ProgramFlags>1</ProgramFlags>
  <FileName>1182_20130121220100.mpg</FileName>
  <HostName>mythserver</HostName>
  <Airdate>2013-01-21</Airdate>
  <Description>When the founder of an adult video franchise is murdered, Beckett and Castle are faced with a lengthy list of suspects; Castle is upset when Alexis starts a revealing video blog.</Description
  <Inetref>82607</Inetref>
  <Season>0</Season>
  <Episode>0</Episode>
  <Channel>
    <ChanId>1182</ChanId>
    <ChanNum>431</ChanNum>
    <CallSign>WZVNDT</CallSign>
    <IconURL/>
    <ChannelName>ABC-HD</ChannelName>
    <MplexId>0</MplexId>
    <TransportId>0</TransportId>
    <ServiceId>0</ServiceId>
    <NetworkId>0</NetworkId>
    <ATSCMajorChan>0</ATSCMajorChan>
    <ATSCMinorChan>0</ATSCMinorChan>
    <Format/>
    <Modulation/>
    <Frequency>0</Frequency>
    <FrequencyId/><FrequencyTable/>
    <FineTune>0</FineTune>
    <SIStandard/>
    <ChanFilters/>
    <SourceId>0</SourceId>
    <InputId>0</InputId>
    <CommFree>0</CommFree>
    <UseEIT>false</UseEIT>
    <Visible>true</Visible>
    <XMLTVID/>
    <DefaultAuth/>
    <Programs/>
  </Channel>
  <Recording>
    <Status>-3</Status>
    <Priority>5</Priority>
    <StartTs>2013-01-22T03:01:00</StartTs>
    <EndTs>2013-01-22T04:00:00</EndTs>
    <RecordId>19</RecordId>
    <RecGroup>Default</RecGroup>
    <PlayGroup>Default</PlayGroup>
    <StorageGroup>Default</StorageGroup>
    <RecType>0</RecType>
    <DupInType>15</DupInType>
    <DupMethod>6</DupMethod>
    <EncoderId>0</EncoderId>
    <Profile>Default</Profile>
  </Recording>
   <Artwork>
      <ArtworkInfos/>
      </Artwork>
</Program>
```

But if I look at Channel table in the DB - all fields are there.
Any other ideas?? What do you get when you do the same call??

**Comment by billmeek** [comment]
@dmfrey, can you explain what happens in this case, where existing recordings
no longer match what's in Channel/GetChannelInfoList?SourceID=X ?

My sense is that users (especially cable users) that have their channels
redefined will loose some/all of their recordings if getting them depends
on matching what's in the channel table on the backend.

A very common solution used on the mythtv-users mailing list is to
tell users to delete all cards, inputs and sources and then re-input
them. So someone added and removed cards in the past might
have chanid like 6xxx and after deleting the above have chanid 
in the 1xxx range.

@sleblanc I recorded that show too and did this: 

```
http://mc0:6544/Dvr/GetRecorded?StartTime=2013-01-22T03:01:00&ChanId=1071
```

and did a dif -w mine yours, the output is here:

```
http://paste.ubuntu.com/1560394/
```

There are a couple of typos, but nothing obviously wrong.

**Comment by billmeek** [comment]
@sleblanc, one other suggestion, that I've used when testing.

Delete the app, verify that all files are gone with My Files:

```
/Root/Android/data/org.mythtv shouldn't exist
```

Then reboot the device and reload 1.0.4.

**Comment by sleblanc** [comment]
Thanks again.

So ... did exactly as you suggested. uninstall, clean, reboot, and reinstall.
Server discovered as my Home server.
Upcoming, Recording Rules, and Guide all work (as they did before).
Recordings - still the same - shows no recordings.

I agree that the data looks the same. 

Could it be a very subtle difference in MythTV versions (since I'm running 0.25)?

Also - I looked at the cache file on my tablet - and it looks like the Recordings data is at least getting there.
Could it be timezone related somehow? I noticed all the dates in the API are UTC. DB times art still local.

**Comment by dmfrey** [comment]
Bill,

I don't think so.  The channels get regularly updated as well.  So if
changes are deleted/moved/added, they will get updated (no matter what it
is), then the next recorded up should just line up.

Also, if a recording comes down, that has no physical channel tied to it,
and the recording has channel details from when it was recorded, then the
channel is added by the recorded load process.  I have to do this, for
instance, for the channels that dummy added for mirobridge to work.

Dan

On Tue, Jan 22, 2013 at 2:16 PM, billmeek notifications@github.com wrote:

> @dmfrey https://github.com/dmfrey, can you explain what happens in this
> case, where existing recordings
> no longer match what's in Channel/GetChannelInfoList?SourceID=X ?
> 
> My sense is that users (especially cable users) that have their channels
> redefined will loose some/all of their recordings if getting them depends
> on matching what's in the channel table on the backend.
> 
> A very common solution used on the mythtv-users mailing list is to
> tell users to delete all cards, inputs and sources and then re-input
> them. So someone added and removed cards in the past might
> have chanid like 6xxx and after deleting the above have chanid
> in the 1xxx range.
> 
> @sleblanc https://github.com/sleblanc I recorded that show too and did
> this:
> 
> http://mc0:6544/Dvr/GetRecorded?StartTime=2013-01-22T03:01:00&ChanId=1071
> 
> and did a dif -w mine yours, the output is here:
> 
> http://paste.ubuntu.com/1560394/
> 
> There are a couple of typos, but nothing obviously wrong.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/MythTV-Clients/MythTV-Android-Frontend/issues/107#issuecomment-12560863.

## 

My Google Profile http://www.google.com/profiles/dmfrey

**Comment by billmeek** [comment]
@sleblanc, I really appreciate you taking the time to test this. And,
after asking you to uninstall/remove data/reboot you can 
tell I'm leaving the realm of scientific trouble shooting.

I know that there were folks on 0.25 with this app working and I
suspect that if it were a 0.25 issue, we'd have heard about it.

If you want to contact me off-list, spell my name backwards and
use gmail. I have some tools on my system that can capture more
data - BUT - you'd have to setup port forwarding on your router
and let me in on 6544. I completely understand if you'd prefer not
to do that.

**Comment by sleblanc** [comment]
Sent email - debug away!!

Thanks again
Steve

**Comment by billmeek** [comment]
@dmfrey, here's a copy of LogCat for Steve's and My entries to the
Recordings page. His fails, mine works. He's on 0.25, I'm on 0.27-pre2.
The two major changes I can think of are Etags and UTC.

Steve's Recordings page: http://paste.ubuntu.com/1561119/
My Recordings page: http://paste.ubuntu.com/1561265/

On line 508 of mine I see: I/RecordedDownloadService(30466): download : GetRecordedList returned 200 OK

But there is no such line in his.

Note that there are no "insert : enter" lines in his.

I know from 5 days ago, he's getting this in his backend log:

```
Jan 17 23:14:38 mythserver mythbackend[22939]: I HttpServer42 servicehost.cpp:281 (ProcessRequest) ServiceHost::ProcessRequest: GetRecordedList : GET /Dvr/GetRecordedList HTTP/1.1#015
Jan 17 23:14:38 mythserver mythbackend[22939]: I HttpServer42 httprequest.cpp:241 (SendResponse) HTTPRequest::SendResponse(xml/html) () :200 OK -> 192.168.0.109: 4
```

**Comment by billmeek** [comment]
In title NCIS- LosAngles/subtitle Blye, K., Part 2" recording, the
<EndTime/> is empty and a NullPointerException is fired. There
could be others, but not saving the endTime 'fixes' the problem.

**Comment by billmeek** [comment]
Changed label to bug. The root cause of this issue was invalid data in the 
user's mythconverg, resulting in an endtime return of:  <EndTime/>.

The above caused an NPE and the Recordings screen never loaded.

Thanks to @sleblanc for his assistance and finding that the error was in
his progend column = '0-0-0 0:0:0' not in endtime, as I had suspected.

The reason for not getting the GetRecordedList returned 200 OK LogCat
entry was due to his being on 0.25.

**Comment by billmeek** [comment]
Reopening for tracking. The original fix handles the case of NULL EndTimes,
however on retesting with NULL StartTimes the app will fire an NPE.

As discussed off-list, flagging episodes like these as a Backend Database
Start/End Time Error would let the user see available information like other
episodes, however, hot play them (let alone fire the NPE.) The Wiki would
be updated to give users more detail on how to find/solve the problem.

**Comment by billmeek** [comment]
Short reply: It appears that once a recording is marked inError, it can't be displayed
when the MythTV DB is repaired (plus a typo that I could fix.)

Retested on: mythtv-android-frontend-1.0.4-63-g78f8fbf

Save good recording progend:

```
SELECT progend FROM recorded WHERE title = 'Chicago Fire' AND subtitle = 'Under the Knife';
```

Created an error with: 

```
UPDATE recorded SET progend = '0-0-0 0:0:0' WHERE title = 'Chicago Fire' AND subtitle = 'Under the Knife' LIMIT 1;
```

Pressed the sync icon on the Recordings screen and received the Toast
message with the: (Recordings has errors) addition.

Restored the progend and verified that no other recordings were in error with:

```
UPDATE recorded SET progend = '2013-02-10 03:00:00' WHERE title = 'Chicago Fire' AND subtitle = 'Under the Knife' LIMIT 1;
SELECT title,progstart FROM recorded ORDER BY progstart DESC;
SELECT title,progend FROM recorded ORDER BY progend DESC;
```

Synced the recordings again, and the (Recordings has errors) message still appeared.

**Comment by dmfrey** [comment]
I was trying to follow this back through the thread...is the problem here that once a recording is marked in error, it can never be unmarked?

**Comment by billmeek** [comment]
Correct, once marked, I couldn't unmark it after the MythTV DB had been corrected.
I started retesting last night and couldn't see the Toast message used before. I'll take
a look in the MAF db. Ball's in my court.

**Comment by billmeek** [comment]
Short answer: null or all 0 timestamps will create one entry in the MAF DB
for each refresh of the Recordings screen. No warning to the user that the
data is bad. If corrected, the episode can't be played (tested on a master BE.)

I think the question becomes how should the user be alerted to the problem.
I'd prefer to see a comment on the Recordings screen, rather than a short
Toast message. Also, remember we have exactly one instance of this problem.
so we're not in the 'show stopper' mode here.

Details:

Running on: mythtv-service-api-0.0.32-93-g0a19bbd

I set progstart and starttime to '' in the MythTV DB. Then did: 

```
http://mc1:6544/Dvr/GetRecordedList?TitleRegEx=Suits
```

In the response they looked like: 

```
<StartTs/> and <StartTime/>
```

I repeated the test with the times set to: '0000-00-00 00:00:00' with the
same results as above.

In the MAF DB below, note that there are 6 entries for the show Suits. But
there is only one in MythTV DB. I believe one entry for each refresh I did
on the Recordings screen.

The IN_ERROR column is set for all but one of the lines. Most likely due
to my restoring the timestamps in the MythTV DB and refreshing again.

There were no Toast messages telling me the recording had bad data. The
time stamp on the Recording screen is that of the last refresh of that screen.

Trying to play the recording fails. This test was done on a master,not a slave.

<pre>
sqlite> .mode column
sqlite> select title,start_time,end_time,in_error from recorded where title='Suits';
Suits       1377883575060  1377883575060  1         
Suits       1377883733832  1377883733832  1         
Suits       1377883752434  1377883752434  1         
Suits       1377884088320  1377884088320  1         
Suits       1377655260000  1377658920000  0         
Suits       1377884656704  1377884656704  1
</pre>
